### PR TITLE
Updates to Excel-related scripts

### PR DESCRIPTION
**Pull request by aoifem01**
Streamlining write_excel scripts to have seasonal-dependent expressions so that we have 1 script instead of autumn/spring versions.

Generally updating other code.

### CODE REVIEW

File: code/write_excel/95_Source_Excel_4.R [hunk 1]
```diff
@@ -24,13 +23,13 @@ year_xx <- year(cut_off_date)
 year_ww <- year_xx - 1
 year_vv <- year_xx - 2
 year_yy <- year_xx + 1
-year_3 <- year_xx - 3
-year_5 <- year_xx - 5
+year_uu <- year_xx - 3
```
Comment (karen-hotopp) [inline code/write_excel/95_Source_Excel_4.R:26]: I did think about sticking to my alphabet system, but then was too lazy to figure out x-5...

File: code/write_excel/95_Source_Excel_4.R [hunk 2]
```diff
@@ -51,45 +50,43 @@ rate_4_1yr <- tail(kpi_4_1yr, n = 3)
...
+  season,
+  {paste0("Data for year ending 31 March ", year_xx, " scheduled to ",
+          "be published in April ", year_yy, " (final data will be ",
+          "produced from data extracted for PHS in September ",
+          year_xx, ").")}, # spring
+  {paste0("KPI data for year ending 31 March ", year_xx, " and some ",
+          "supplementary information are planned for publication in April ", year_yy)} # autumn
+)
 qpmg_review <- paste0("For review at QPMG in ", qpmg_month, " ", year_xx)
 tab_vasc_desc <- paste0("Vascular KPIs background information: Vascular referral ",
```
Comment (karen-hotopp) [inline code/write_excel/95_Source_Excel_4.R:66]: TO REVIEW

**Review by karen-hotopp — COMMENTED** [review]
Several comments here and there to address. I think some things should be recorded for a future branch/pull request. Some issues with the printing out to Excel, but I think that may be because of templates I used.

File: code/4_3_kpi_2.R [hunk 3]
```diff
@@ -827,6 +831,7 @@ qa_detail <- bind_rows(summary_text, summary_detail, summary) %>%
          year3_p)
 
 # Insert any missing detail categories and arrange
+# AMc note - this would be better further up probably
```
[the remark this answers is not in the document]
Comment (aoifem01) [reply code/4_3_kpi_2.R:834]: actually doesn't work further up, left as is

File: code/4_3_kpi_2.R [hunk 4]
```diff
@@ -1000,11 +1006,26 @@ kpi_2 <- bind_rows(kpi_2_1a, kpi_2_1b, kpi_2_2, kpi_2_2_add_a, kpi_2_2_add_b,
                    qa_batch_scot, qa_batch_hb, qa_batch_recall) %>% 
   rename(fin_year = financial_year,
          hbres = hb_screen) # done to make formatting easier in Write Excel
```
[the remark this answers is not in the document]
Comment (aoifem01) [reply code/4_3_kpi_2.R:1008]: leaving additional device groups out, as I'm not certain the QPMG members want it long-term

File: code/5_4_kpi_3.R [hunk 5]
```diff
@@ -446,17 +454,20 @@ table(kpi_3$kpi)
 ## Historical file
 # Create backup of last year's file
 hist_db <- read_rds(paste0(hist_path,"/aaa_kpi_historical_theme4.rds"))
+table(hist_db$kpi, hist_db$financial_year)
+
 
 # temp: renaming "financial_year" to "fin_year" to make below function work
 # AMc note: discuss with KH as to whether this can be changed permanently??
```
[the remark this answers is not in the document]
Comment (aoifem01) [reply code/5_4_kpi_3.R:461]: I haven't made it a permanent change in that the current historical table still has it as "financial_year" - but I have made a new version of it with this changed which is ready for this next kpi run, just need to have you double check it's just the name I've changed - will also need to add in a SIMD column

File: code/write_excel/93_Source_Excel_2.R [hunk 6]
```diff
@@ -269,18 +256,17 @@ kpi_1.4b_head1 <- paste0("Due to attend quarterly surveillance in year ending 31
...
 
 ## Table 6: Surveillance Notes ----
 ### table headers ----
 table6_head1 <- paste0("Screened in year ending 31 March",  '\n',
                        year_vv)
 table6_head2 <- paste0("Screened in year ending 31 March",  '\n',
                        year_ww)
-table6_head3 <- paste0("Screened from 1 April ", year_ww, " - 28",
-                       '\n', "February ", year_xx)
+table6_head3 <- paste0("Screened in year ending 31 March",  '\n',
```
[the remark this answers is not in the document]
Comment (aoifem01) [reply code/write_excel/93_Source_Excel_2.R:268]: I believe I changed it to what we had agreed in the last meeting

**Comment by aoifem01** [comment]
New commits above (b379bdd) to address majority of review comments

**Review by karen-hotopp — APPROVED** [review]

**Pull request merged**